Packs/MicrosoftGraphMail/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py

```diff
@@ -693,7 +693,34 @@ def get_emails(self, exclude_ids, last_fetch, folder_id):
             "$top": len(exclude_ids) + self._emails_fetch_limit  # fetch extra incidents
         }
 
-        return self.ms_client.http_request('GET', suffix_endpoint, params=params).get('value') or []
+        emails_as_html = self.ms_client.http_request('GET', suffix_endpoint, params=params).get('value') or []
+
+        headers = {
+            "Prefer": "outlook.body-content-type='text'"
+        }
+
+        emails_as_text = self.ms_client.http_request(
+            'GET', suffix_endpoint, params=params, headers=headers
+        ).get('value') or []
+
+        return self.get_emails_as_text_and_html(emails_as_html=emails_as_html, emails_as_text=emails_as_text)
+
+    @staticmethod
+    def get_emails_as_text_and_html(emails_as_html, emails_as_text):
+
+        text_emails_ids = {email.get('id'): email for email in emails_as_text}
+        emails_as_html_and_text = []
+
+        for email_as_html in emails_as_html:
+            body_as_text = text_emails_ids.get(email_as_html.get('id'), {}).get('body')
```

Comment:
lets extract text_emails_ids.get(email_as_html.get('id') to email_as_text, esp since its used twice